pandas/io/formats/style.py

```diff
@@ -1444,26 +1517,33 @@ def _background_gradient(
         text_color_threshold: float = 0.408,
         vmin: Optional[float] = None,
         vmax: Optional[float] = None,
+        gmap: Optional[Sequence] = None,
     ):
         """
-        Color background in a range according to the data.
+        Color background in a range according to the data or a gradient map
         """
-        if (
-            not isinstance(text_color_threshold, (float, int))
-            or not 0 <= text_color_threshold <= 1
-        ):
-            msg = "`text_color_threshold` must be a value from 0 to 1."
-            raise ValueError(msg)
+        if gmap is None:  # the data is used the gmap
+            gmap = s.to_numpy(dtype=float)
+        else:  # gmap is conformed to the data shape
+            if isinstance(gmap, (Series, DataFrame)):  # align indx / cols
```

Comment:
do any of these cases interact with `axis=` meaning if you supply a Dataframe but axis=1 does this work? (or conversley is axis=None the *only* case where a DataFrame / 2D accepted?)

Reply:
@jreback I have added an arg validation function. I think this is better, and it is also to preempt similar reservations you may have had in #39821 where I am now reusing the same code.